core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/internal/util/ParserToolkit.java

```diff
@@ -70,25 +81,35 @@ public static void assertValue(Object value, Object ... accepted) throws Invalid
 		throw new InvalidJfrFileException(value + " is not among the expected values"); //$NON-NLS-1$
 	}
 
+	/*
+	 * A helper cache for the unrecognized frame types to reduce the amount of allocated instances.
+	 * The expectation is that the number of unrecognized frame types will be very small, usually
+	 * zero, so the memory overhead of the cache stays negligible.
+	 */
+	private static final Map<String, IMCFrame.Type> TYPE_CACHE = new HashMap<>();
```

Comment:
I would suggest to move the Type Cache to be part of the Type implementation in IMCFrame.